lib/configurators/aws/aws.go

```diff
@@ -1066,7 +1117,7 @@ func getTargetConfig(flags configurators.BootstrapFlags, cfg *servicecfg.Config,
 // awsMatchersFromConfig is a helper function that extracts database AWS matchers
 // from the service configuration based on cli flags.
 func awsMatchersFromConfig(flags configurators.BootstrapFlags, cfg *servicecfg.Config) []types.AWSMatcher {
-	if flags.DiscoveryService {
+	if flags.DiscoveryService || flags.DiscoveryServiceConfig {
```

Comment:
Does th escenario where `flags.DiscoveryService == false`  `flags.DiscoveryServiceConfig==true` is even possible ?

Reply:
No. Technically `flags.DiscoveryServiceConfig==true` should be enough but since `flags.DiscoveryService` check was there before so i kept it there just to be safe. I can remove `flags.DiscoveryService` check if it's cleaner.

Alternatively, I can combine the two booleans to an enum
```
type TargetService int                                
const (                                
    DatabaseService TargetService = iota                                
    DiscoveryService                                
    DatabaseServiceWithDiscoveryConfig                                
)   
```
This could be more future proof but there are a lot more places need to be updated.

what do you think?

Reply:
I think it's possible when `teleport discovery bootstrap --database-service-role="..."` is used, because of this code:

https://github.com/gravitational/teleport/blob/2b39df8fd1853ed314c61d7966827afa87c0e203/tool/teleport/common/configurator.go#L123-L149

Reply:
Also, I think when that flow is used the policy document generated for database service auth will have this description:
> Used by Teleport database agents for discovering AWS-hosted databases.

But that should probably be changed if the policy is just for authorization.

Coming from this code:

https://github.com/gravitational/teleport/blob/2b39df8fd1853ed314c61d7966827afa87c0e203/lib/configurators/aws/aws.go#L634-L649

Reply:
+1 for making an enum or something, just to decide and clarify early on in the code what the configurator is trying to do:
1. configuring discovery service
2. configuring database service discovery + authz (legacy)
3. configuring database service for authz only (using the discovery service section)